File: 1458-sort-integers-by-the-number-of-1-bits/sort-integers-by-the-number-of-1-bits.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var sortByBits = function(arr) {
    
    

    function countBits(num){
       let cnt = 0;
        while (num) {
            num = num & (num - 1);
            cnt++;
        }
        return cnt;

    }
    arr.sort((a,b)=>{
        const aBit =countBits(a) 
        const bBit = countBits(b)
        if( aBit!==bBit){

            return aBit-bBit
        }


        return a-b;
    })
    return arr; 
};

File: 1458-sort-integers-by-the-number-of-1-bits/sort-integers-by-the-number-of-1-bits.py
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:

        def num_bits(num):

            cnt = 0
            while(num):
                num =  num &( num-1)
                cnt+=1
            return cnt

        
        arr.sort(key = lambda x: (num_bits(x),x))   
        return arr   